cms/djangoapps/contentstore/models.py

```
@@ -157,6 +171,13 @@ def upstream_context_title(self) -> str:
         """
         return self.upstream_block.publishable_entity.learning_package.title
 
+    @property
+    def published_at(self) -> str | None:
+        """
+        Returns the published date of the component
+        """
+        return self.upstream_block.publishable_entity.published.publish_log_record.publish_log.published_at
```

Do we need to check if `self.upstream_block.publishable_entity.published` is `None`? I'm just wondering if this would throw an error instead of returning `None` if used on an entity that was never published. (Which could happen if you copy-paste a draft.)

---

Updated: https://github.com/openedx/edx-platform/pull/37208/commits/7751be4bb47c241fba3ce7a08e24fa271dc0b659